hexl/thread-pool/thread-pool-executor.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "hexl/thread-pool/thread-pool-vars.hpp"
+#include "thread-pool/thread-pool.hpp"
+
+namespace intel {
+namespace hexl {
+
+class ThreadPoolExecutor {
+#ifdef HEXL_MULTI_THREADING
+
+ private:
+  inline static ThreadPool* pool = new ThreadPool();
```

Comment:
```suggestion
  static ThreadPool* pool = new ThreadPool();
```
inline keyword not required in class body as it is implied

Reply:
If I remove inline I get the following error:
 error: ‘constexpr’ needed for in-class initialization of static data member ‘intel::hexl::ThreadPool* intel::hexl::ThreadPoolExecutor::pool’ of non-integral type [-fpermissive]
Then I get reference error if initializing the member somewhere else

Reply:
my bad that one is not a method

Reply:
ThreadPoolExecutor is never instantiated. So it's not like it can be created on constructor. If I check whether (pool == nullptr) on SetNumberOfThreads, AddParallelJobs and AddRecursiveCalls I could add it once at that moment. Then (pool == nullptr) will always be false.